store.hpp

```diff
@@ -0,0 +1,176 @@
+#pragma once
+#include <list>
+#include <map>
+
```

Comment:
You didn't include `vector` and `string` here, so it gave me compile errors that the header files are missing.